src/mono/wasm/runtime/startup.ts

```diff
@@ -329,6 +331,10 @@ async function mono_wasm_after_user_runtime_initialized(): Promise<void> {
 
         if (runtimeHelpers.diagnosticTracing) console.debug("MONO_WASM: Initializing mono runtime");
 
+        if (MonoWasmThreads) {
```

Comment:
maybe you could now move it earlier in the startup, just after `beforeOnRuntimeInitialized.promise_control.resolve`. but I'm not sure it's useful.

Reply:
I'll move it. the only hard requirement is that the diagnostic server has to start before we call into driver.c to start the root domain initialization.  Starting later is better

Reply:
Ah, there's an annoying problem here.  In the non-Blazor startup we don't set environment variables until `_apply_configuration_from_args`.  So actually I have to try to init diagnostics in two places.